framework/src/actions/ActionWarehouse.C

```diff
@@ -349,8 +349,12 @@ ActionWarehouse::printInputFile(std::ostream & out)
   for (std::vector<Action* >::iterator i = ordered_actions.begin();
        i != ordered_actions.end();
        ++i)
-   {
-    std::string name ((*i)->name());
+  {
+    std::string name;
+    if ((*i)->isParamValid("parser_syntax"))
+      name = (*i)->getParam<std::string>("parser_syntax");
+    else
+      name = (*i)->name();
```

Comment:
We haven't really watched the changes to the input file printing but this is a fairly big impact. This means that any user created object won't fit into the right place in the tree. Have you looked at the new `--show-input` output after this change?